### PR TITLE
Gamedev

### PR DESCRIPTION
**Pull request by tgomudur**
Changelog:
- Resourcing sharing between agent and player. (Game V1)
- Agent gameplay and resource update on side panel
- Includes sponsor requested changes like providing hint to the player.
- Updating user statistics once game is completed.
- Minor bug fixes
- Reformatting and code cleanup

### CODE REVIEW

**Review by yousuf031991 — APPROVED** [review]
+1

**Comment by adityab17190** [comment]
There are some old commits showing up in this PR.

**Comment by adityab17190** [comment]
+1 for the user stats query update

**Comment by super9user** [comment]
Yes, there are many old commits which seems to be unnecessary. 

Also, pls do not remove .gitkeep from the public/downloads folder.

**Comment by tgomudur** [comment]
I have added the gitkeep file in public/downloads folder. I cannot do anything about the old commits for now without rebasing and applying all the patches. Resolving all the conflicts for those patches will be an effort not worth doing. Since the game is so far behind in the master, I want to merge it ASAP. If you have any more comments please let me know, otherwise I'll merge it tomorrow.

**Review by super9user — APPROVED** [review]

File: MainApp/run.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+mongod --dbpath=/media/hospmanagement/My\ Book/MachineLearning/HumanAgentInteraction/MainApp/database> /dev/null & npm run build && pm2 start dist/index.js
```
Comment (super9user) [inline MainApp/run.sh:1]: Thank you for doing this.
Can you add pm2 in package.json?

Comment (tgomudur) [reply MainApp/run.sh:1]: Sorry it took so long. I added it now.

**Pull request merged**